multinet/api/tasks/upload/csv.py

```diff
@@ -67,3 +69,129 @@ def process_csv(
 
     # Insert rows
     table.put_rows(csv_rows)
+
+
+def create_csv_network(workspace: Workspace, serializer):
+    """Create a network from a link of tables (in request thread)."""
+    from multinet.api.views.serializers import CSVNetworkCreateSerializer
+
+    serializer: CSVNetworkCreateSerializer
+    serializer.is_valid(raise_exception=True)
+
+    # Perform joins before edge creation, since new tables are created
+    mapped_tables = {}
+    for table, mapping in serializer.validated_data.get('joins', {}).items():
+        foreign_table = mapping['foreign_column']['table']
+        foreign_column = mapping['foreign_column']['column']
+        joined_table, created = Table.objects.get_or_create(
+            name=f'{foreign_table}-joined-{table}', workspace=workspace
+        )
+
+        # Clear table rows if already exists
+        joined_table: Table
+        mapped_tables[foreign_table] = joined_table.name
+        if created:
+            joined_table.get_arango_collection(readonly=False).truncate()
```

Comment:
Is it, generally speaking, ok to do this? The edge case is where someone had a weirdly named table by that joined name already. It is probably not a problem for us, especially in the short term, but I wonder if an error condition would be more appropriate?

Reply:
I think it's okay, but it'd be pretty hard to catch this error here, since there'd be no way to distinguish between a table by that name created by a user, and one created through a normal join. We could disallow creating tables with names that include `-joined-` though.

Reply:
You can't just check for a clashing table name before creating it?

Reply:
My initial reason for doing this was in case someone created a network with a join, but then decided to change some underlying data and re-created a network with the same join again, thus not having multiple tables with essentially the same data.

We _could_ check for clashing table names, but the action we should take from there is ambiguous imo (since then this check for existing table name would become much more complicated, we'd have to come up with a new name, etc.).

I think we should either:
1. Just keep things as they are, ignoring potential names clashes
2. If you don't like the potential name clash, we should just do away with trying to re-use these joined tables, and just create unique joined table names every time.

What do you think?

Reply:
I see, you're trying to reuse these automated join tables.

I think we're fine for now, but that disallowing manually created tables with `-joined-` in the name would be reasonable. For now, if you just log an issue tracking this problem, that should be fine.

Reply:
This has been filed as #129. Thanks @AlmightyYakob!